src/features/posts/selectedPostSlice.ts

```diff
@@ -0,0 +1,18 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+
+const initialState = null as Post | null;
```

Comment:
Consider specifying the type of `initialState` more explicitly for better type-checking. You can use `Post | null` directly instead of using `as`. This can help with readability and maintainability of the code.